packages/integrations/cloudflare/src/index.ts

```diff
@@ -90,33 +94,91 @@ export default function createIntegration(args?: Options): AstroIntegration {
 					vite.ssr.target = 'webworker';
 				}
 			},
+			// BUG: `entrypoints` currently does not respect SERVER_BUILD_FOLDER, aka `build.server`
+			'astro:build:ssr': ({ manifest, entryPoints }) => {
+				// NOTE: monkey-patch because of entrypoints bug:
+				const pathToReplace = fileURLToPath(new URL(_config.outDir));
+				newEntryDir = fileURLToPath(new URL(_buildConfig.server, _config.outDir));
+
+				entryPoints.forEach((value, key) => {
+					_entryPoints.set(key, new URL(value.href.replace(pathToReplace, newEntryDir)));
```

Comment:
I'm guessing this is what is making Windows fail. I would recommend keeping everything as file URLs as long as possible, it prevents this sort of thing from happening.

Can you explain what this code is doing? It looks like it is adding the buildConfig.server to the outDir. Is that not already there?

Reply:
it remaps the entryPoints to the correct folder, because the `URL` for the entryPoints do not take into account `server: new URL(`.${SERVER_BUILD_FOLDER}`, config.outDir),`

Give the file which is available at `astro:build:ssr`, the exposed entryPoint does not return the correct path AFAIK. Notice the difference: dist/**$server_build**/src/pages/_person_/entry._car_.astro.mjs

```
real file path: packages/integrations/cloudflare/test/fixtures/split/dist/$server_build/src/pages/_person_/entry._car_.astro.mjs

object returned by entryPoints: {
    route: '/[person]/[car]',
    type: 'page',
    pattern: /^\/([^/]+?)\/([^/]+?)\/?$/,
    segments: [ [Array], [Array] ],
    params: [ 'person', 'car' ],
    component: 'src/pages/[person]/[car].astro',
    generate: [Function (anonymous)],
    pathname: undefined,
    prerender: false
  } => URL {
    href: 'file:///Users/alexanderniebuhr/Developer/Projects/alexanderniebuhr_astro/packages/integrations/cloudflare/test/fixtures/split/dist/src/pages/_person_/entry._car_.astro.mjs',
    origin: 'null',
    protocol: 'file:',
    username: '',
    password: '',
    host: '',
    hostname: '',
    port: '',
    pathname: '/Users/alexanderniebuhr/Developer/Projects/alexanderniebuhr_astro/packages/integrations/cloudflare/test/fixtures/split/dist/src/pages/_person_/entry._car_.astro.mjs',
    search: '',
    searchParams: URLSearchParams {},
    hash: ''
  }
```

Reply:
_note to self: should be fixed by https://github.com/withastro/astro/pull/7483_